tools/mo/unit_tests/mo/back/ie_ir_ver_2/emitter_test.py

```diff
@@ -117,3 +126,143 @@ def test_serialize_old_api_map_result(self):
+            **shaped_parameter('indices', int64_array([1, 2]), {'data_type': np.float32, 'type': Parameter.op}),
+            **valued_const_with_data('axis', int64_array(1)),
+            **regular_op_with_empty_data('gather', {'op': 'Gather', 'batch_dims': 0, 'infer': Gather.infer,
+                                                    'type': Gather.op}),
+            **result('res'),
+        }
+
+        edges = [
+            *connect('data', '0:gather'),
+            *connect('indices', '1:gather'),
+            *connect('axis', '2:gather'),
+            *connect('gather', 'res'),
+        ]
+
+        graph = build_graph(nodes, edges)
+
+        data_node = Node(graph, 'data')
+        Parameter.update_node_stat(data_node, {})
+        indices_node = Node(graph, 'indices')
+        Parameter.update_node_stat(indices_node, {})
+
+        gather_node = Node(graph, 'gather')
+        Gather.update_node_stat(gather_node, {})
+
+        res_node = Node(graph, 'res')
+        Result.update_node_stat(res_node, {})
+
+        partial_infer(graph)
+        type_infer(graph)
+
+        return graph
+
+    @staticmethod
+    def build_graph_with_maxpool():
+        graph = build_graph(
+            nodes_attrs={
+                'input': {'kind': 'op', 'op': 'Parameter', 'name': 'node', 'infer': Parameter.infer,
+                          'shape': [1, 3, 10, 10]},
+                'input_data': {'kind': 'data', 'value': None, 'shape': None},
+
+                'pool': {'kind': 'op', 'type': 'MaxPool', 'infer': Pooling.infer,
+                         'window': np.array([1, 1, 2, 2]), 'stride': np.array([1, 1, 2, 2]),
+                         'pad': np.array([[0, 0], [0, 0], [0, 0], [1, 1]]),
+                         'pad_spatial_shape': np.array([[0, 0], [1, 1]]),
+                         'pool_method': 'max', 'exclude_pad': False, 'global_pool': False,
+                         'output_spatial_shape': None, 'output_shape': None,
+                         'kernel_spatial': np.array([2, 2]), 'spatial_dims': np.array([2, 3]),
+                         'channel_dims': np.array([1]), 'batch_dims': np.array([0]),
+                         'pooling_convention': 'full', 'dilation': np.array([1, 1, 2, 2]),
+                         'auto_pad': 'valid'},
+
+                'pool_data': {'kind': 'data', 'value': None, 'shape': None},
+                'pool_data_added': {'kind': 'data', 'value': None, 'shape': None},
+                'result': {'kind': 'op', 'op': 'Result'},
+                'result_added': {'kind': 'op', 'op': 'Result', 'infer': lambda x: None}
+            },
+            edges=[
+                ('input', 'input_data'),
+                ('input_data', 'pool'),
+                ('pool', 'pool_data'),
+                ('pool_data', 'result'),
+                ('pool', 'pool_data_added'),
+                ('pool_data_added', 'result_added')
```

Comment:
Not sure that this is valid case at all that for the same output port we have several data nodes.

Reply:
`build_graph` automatically assigns port numbers if they are not specified explicitly. And i saw that ports when dumped for graphviz
https://github.com/openvinotoolkit/openvino/blob/1db4446e2a6ead55d066e0b4e718fa37f509353a/tools/mo/unit_tests/utils/graph.py#L176-L179

Such creation of test graphs is done in many places, e.g. here is maxpool normalizer from where i copied this fragment
https://github.com/openvinotoolkit/openvino/blob/ff6a9a117991be4c9736b4935d579ee2cdeae39b/tools/mo/unit_tests/mo/back/MaxPoolTest.py#L79-L86

But anyway, i agree, explicitly is better. I added explicit port setting